### ISSUE TITLE
Encoding issue (XeroError: signature_invalid) when downloading attachment with filename containing a ","

### ISSUE THREAD

**Issue by o1iver** [XeroAPI/xero-node#281]
_Note: this may actually be an issue with the API server itself (something to do with IIS)._

# Reproducing

See example client at: https://github.com/o1iver/xero-issue

## Setup Xero

Create a trial organization and add a single bill to pay. Attach the following two PDFs to this bill:

- [GoodPdf.pdf](https://github.com/o1iver/xero-issue/blob/master/GoodPdf.pdf)
- [Bad,Pdf.pdf](https://github.com/o1iver/xero-issue/blob/master/Bad%2CPdf.pdf)

![image](https://user-images.githubusercontent.com/282765/64628363-8f85cb00-d3f1-11e9-908f-1c191e0ff0dc.png)

Setup private app, i.e. get secret and key and add to config.json as shown in [README](https://github.com/o1iver/xero-issue/blob/master/README.md).

## Try private client app

Run `npm install` and `npm start`.

# Result

```bash
"IsDiscounted": false,
  "HasAttachments": true,
  "HasErrors": false,
  "Contact": {
    "ContactID": "62be196e-1013-4116-b3d7-a460a8b384a9",
    "Name": "Example",
    "Addresses": [],
    "Phones": [],
    "ContactGroups": [],
    "ContactPersons": [],
    "HasValidationErrors": false
  },
  "DateString": "2019-09-10T00:00:00",
  "Date": "/Date(1568073600000+0000)/",
  "DueDateString": "2019-09-10T00:00:00",
  "DueDate": "/Date(1568073600000+0000)/",
  "Status": "AUTHORISED",
  "LineAmountTypes": "Exclusive",
  "LineItems": [],
  "SubTotal": 1,
  "TotalTax": 0,
  "Total": 1,
  "UpdatedDateUTC": "/Date(1568127617290+0000)/",
  "CurrencyCode": "EUR"
}
{
  "AttachmentID": "2248e799-3da6-4125-9220-ed9f07ebf2f9",
  "FileName": "GoodPdf.pdf",
  "Url": "https://api.xero.com/api.xro/2.0/invoices/d0939191-0568-4672-9046-200ce738f458/Attachments/GoodPdf.pdf",
  "MimeType": "application/pdf",
  "ContentLength": 30112
}
{
  "AttachmentID": "97749a0d-dd16-4a58-96e3-d6f02538515e",
  "FileName": "Bad,Pdf.pdf",
  "Url": "https://api.xero.com/api.xro/2.0/invoices/d0939191-0568-4672-9046-200ce738f458/Attachments/Bad%2CPdf.pdf",
  "MimeType": "application/pdf",
  "ContentLength": 30112
}
Succeeded downloading 'GoodPdf.pdf'.
(node:4570) UnhandledPromiseRejectionWarning: Error: XeroError: signature_invalid (Failed to validate signature)
    at new XeroError (/mnt/c/Users/Oliver/xero-issue/node_modules/xero-node/lib/XeroError.js:33:24)
    at /mnt/c/Users/Oliver/xero-issue/node_modules/xero-node/lib/internals/OAuth1HttpClient.js:170:49
    at passBackControl (/mnt/c/Users/Oliver/xero-issue/node_modules/xero-node/lib/internals/OAuth1HttpClient.js:227:21)
    at IncomingMessage.<anonymous> (/mnt/c/Users/Oliver/xero-issue/node_modules/xero-node/lib/internals/OAuth1HttpClient.js:236:21)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:4570) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)(node:4570) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by iamam34** [comment]
This is probably related to the way we escape the filename in xero-node. https://github.com/XeroAPI/xero-node/blob/master/src/AccountingAPIClient.ts#L48

**Comment by RettBehrens** [comment]
A quick note on why we’re closing this issue:

With the release of version 4.x of the Xero-Node SDK in December of 2019, we delivered a complete overhaul of the SDK. 

We moved from writing our SDKs by hand to generating our SDKs using our [Xero-OpenAPI](https://github.com/XeroAPI/Xero-OpenAPI) specs. By utilizing Xero-OpenAPI specs to generate code we’ve improved the developer experience of maintaining the SDK and improved the accessibility for our developer community to contribute via pull requests to the Xero-OpenAPI specs. 

We also built our new SDKs to use the industry standard OAuth2 in preparation for our March 2021 deprecation of OAuth1. For more details, see our [blog post on our decision to move to OAuth2](https://devblog.xero.com/an-update-on-why-we-are-saying-goodbye-oauth-1-0a-hello-oauth-2-0-6a839230908f). 

Version 3.x using Oauth1 and version 4.x using OAuth2 do not share any code as we now generate our SDKS. Going forward we’re focusing on building, maintaining and improving our generated SDKs but you are welcome to fork the [‘oauth1’ branch](https://github.com/XeroAPI/xero-node/tree/oauth1) and apply changes if you wish to implement something for your integration.
